### DISCUSSION TITLE
How to split service registration?

### DISCUSSION THREAD

**Discussion by erikpa1** [actix/actix-web#2926] [Q&A] [answered]
Hi,

I would like to split my code to modules like licensing, projects, users,....

and I would like to give opportunity to modules register themselves to App<AppEntry> but how to do it ? 

I want to achieve something like:

```
fn mount_licencing(app: App<AppEntry>) -> App<AppEntry> {
   app.service(function_a)
       .service(function_b)
       .service(function_c)
}

fn mount_project(app: App<AppEntry>)-> App<AppEntry> {
   app.service(function_a)
       .service(function_b)
       .service(function_c)
}

```

How to achieve this code ? because AppEntry is private, and I cant access it to split code?

**Comment by robjtede** [comment, marked as answer]
[`App::configure`](https://docs.rs/actix-web/latest/actix_web/struct.App.html#method.configure)

**Reply by erikpa1** [reply]
@robjtede thanks I will try
